dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/VerifySignatureUtility.java

```diff
@@ -0,0 +1,364 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.install.internal;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.net.URLConnection;
+import java.security.GeneralSecurityException;
+import java.security.Security;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPCompressedData;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection;
+import org.bouncycastle.openpgp.PGPSignature;
+import org.bouncycastle.openpgp.PGPSignatureList;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentVerifierBuilderProvider;
+
+import com.ibm.ws.install.InstallConstants;
+import com.ibm.ws.install.InstallConstants.VerifyOption;
+import com.ibm.ws.install.InstallException;
+import com.ibm.ws.install.internal.InstallLogUtils.Messages;
+import com.ibm.ws.kernel.boot.cmdline.Utils;
+
+/**
+ *
+ */
+public class VerifySignatureUtility {
+
+    private static final Logger logger = InstallLogUtils.getInstallLogger();
+    private static final String DEFAULT_LIBERTY_KEY_ID = "0x4D210F6946102B8E";
+    private static final String UBUNTU_HOST = "keyserver.ubuntu.com";
+    private static final String PGP_HOST = "keys.openpgp.org";
+    private static final String MIT_HOST = "pgp.mit.edu";
+    private static final String UbuntuServerURL = "https://keyserver.ubuntu.com/pks/lookup?op=get&options=mr&search=";
+    private static final String MITServerURL = "https://pgp.mit.edu/pks/lookup?op=get&options=mr&search=";
+    private static final String PGPServerURL = "https://keys.openpgp.org/pks/lookup?op=get&options=mr&search=";
+
+    private final ProgressBar progressBar = ProgressBar.getInstance();
+
+    VerifySignatureUtility() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    public boolean isLibertyKeyValid() {
```

Comment:
let's add a path parameter to this method which specifies the path to the key being validated. 

Rename it to isKeyValid(Path keyPath), or something so that callers know it can validate any key .